src/test/java/htsjdk/samtools/util/IOUtilUnitTest.java

```diff
@@ -1,397 +1,391 @@
+    private static final File FIVE_SPACES_THEN_A_NEWLINE_THEN_FIVE_SPACES_FILE = new File("src/test/resources/htsjdk/samtools/io/5newline5.txt");
+    private static final List<String> SLURP_TEST_LINES = Arrays.asList("bacon   and rice   ", "for breakfast  ", "wont you join me");
+    private static final String SLURP_TEST_LINE_SEPARATOR = "\n";
+    private static final String TEST_FILE_PREFIX = "htsjdk-IOUtilTest";
+    private static final String TEST_FILE_EXTENSIONS[] = {".txt", ".txt.gz"};
+    private static final String TEST_STRING = "bar!";
+
+    private File existingTempFile;
+    private String systemUser;
+    private String systemTempDir;
+    private FileSystem inMemoryFileSystem;
+
+    @BeforeClass
+    public void setUp() throws IOException {
+        existingTempFile = File.createTempFile("FiletypeTest.", ".tmp");
+        existingTempFile.deleteOnExit();
+        systemTempDir = System.getProperty("java.io.tmpdir");
+        final File tmpDir = new File(systemTempDir);
+        inMemoryFileSystem = Jimfs.newFileSystem(Configuration.unix());;
+        if (!tmpDir.isDirectory()) tmpDir.mkdir();
+        if (!tmpDir.isDirectory())
+            throw new RuntimeException("java.io.tmpdir (" + systemTempDir + ") is not a directory");
+        systemUser = System.getProperty("user.name");
+    }
+
+    @AfterClass
+    public void tearDown() throws IOException {
+        // reset java properties to original
+        System.setProperty("java.io.tmpdir", systemTempDir);
+        System.setProperty("user.name", systemUser);
+        inMemoryFileSystem.close();
+    }
+
+    @Test
+    public void testFileReadingAndWriting() throws IOException {
+        String randomizedTestString = TEST_STRING + System.currentTimeMillis();
+        for (String ext : TEST_FILE_EXTENSIONS) {
+            File f = File.createTempFile(TEST_FILE_PREFIX, ext);
+            f.deleteOnExit();
+
+            OutputStream os = IOUtil.openFileForWriting(f);
+            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(os));
+            writer.write(randomizedTestString);
+            writer.close();
+
+            InputStream is = IOUtil.openFileForReading(f);
+            BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+            String line = reader.readLine();
+            Assert.assertEquals(randomizedTestString, line);
+        }
+    }
+
+    @Test(groups = {"unix"})
+    public void testGetCanonicalPath() throws IOException {
+        String tmpPath = System.getProperty("java.io.tmpdir");
+        String userName = System.getProperty("user.name");
+
+        if (tmpPath.endsWith(userName)) {
+            tmpPath = tmpPath.substring(0, tmpPath.length() - userName.length());
+        }
+
+        File tmpDir = new File(tmpPath, userName);
+        tmpDir.mkdir();
+        File actual = new File(tmpDir, "actual.txt");
+        actual.deleteOnExit();
```

Comment:
You should really consider introducing a `createTempFile()` method in htsjdk that auto-marks things for deletion, so that test code doesn't have to worry about it.